markets/bfp-market/contracts/modules/PerpAccountModule.sol

```diff
@@ -118,4 +136,128 @@ contract PerpAccountModule is IPerpAccountModule {
                 mm
             );
     }
+
+    function realizePosition(
+        PerpMarket.Data storage market,
+        Position.Data storage position,
+        Margin.MarginValues memory marginValues,
+        Margin.Data storage accountMargin,
+        PerpMarketConfiguration.Data storage marketConfig,
+        uint256 oraclePrice
+    ) private {
+        // Determine if toPosition can be immediately liquidated.
+        if (Position.isLiquidatable(position, market, oraclePrice, marketConfig, marginValues)) {
+            revert ErrorUtil.CanLiquidatePosition();
+        }
+
+        accountMargin.updateAccountDebtAndCollateral(
+            market,
+            marginValues.marginUsd.toInt() - marginValues.collateralUsd.toInt()
+        );
+    }
+
+    /**
+     * @inheritdoc IPerpAccountModule
+     */
+    function mergeAccounts(uint128 fromId, uint128 toId, uint128 marketId) external {
+        // Check msg sender is owner for both the accounts.
+        Account.loadAccountAndValidatePermission(
+            fromId,
+            AccountRBAC._PERPS_MODIFY_COLLATERAL_PERMISSION // TODO maybe new permission?
+        );
+        Account.loadAccountAndValidatePermission(
+            toId,
+            AccountRBAC._PERPS_MODIFY_COLLATERAL_PERMISSION // TODO maybe new permission?
+        );
+        PerpMarket.Data storage market = PerpMarket.exists(marketId);
+        PerpMarketConfiguration.Data storage marketConfig = PerpMarketConfiguration.load(marketId);
+        Margin.GlobalData storage globalMarginConfig = Margin.load();
+        Margin.Data storage fromAccountMargin = Margin.load(fromId, marketId);
+
+        Runtime_mergeAccounts memory runtime;
+
+        runtime.supportedSynthMarketIdsLength = globalMarginConfig.supportedSynthMarketIds.length;
+
+        for (uint256 i = 0; i < runtime.supportedSynthMarketIdsLength; ) {
+            runtime.synthMarketIdForLoop = globalMarginConfig.supportedSynthMarketIds[i];
+            runtime.fromAccountCollateralForLoop = fromAccountMargin.collaterals[
```

Comment:
`forLoop` in the var name kinda smells 🤔

Reply:
I do need to keep track of them, renamed to `currentSynthMarketId`